platform/ios/MGLMapboxEvents.m

```diff
@@ -174,17 +175,20 @@ + (instancetype)sharedManager {
     static dispatch_once_t onceToken;
     static MGLMapboxEvents *_sharedManager;
     dispatch_once(&onceToken, ^{
-        void (^setupBlock)() = ^{
-            _sharedManager = [[self alloc] init];
-            // setup dedicated location manager on first use
-            [MGLMetricsLocationManager sharedManager];
-        };
-        if ( ! [[NSThread currentThread] isMainThread]) {
-            dispatch_sync(dispatch_get_main_queue(), ^{
+        if ( ! NSProcessInfo.processInfo.mgl_isInterfaceBuilderDesignablesAgent) {
```

Comment:
That’s correct. Whenever you modify an instance of `MGLMapView` in a storyboard (for instance by resizing it), Interface Builder spawns an IBDesignablesAgentCocoaTouch process in the background that instantiates an `MGLMapView` and calls methods like `-drawRect:` on it. There are two hooks for IB-only code that should never run in front of a user: `TARGET_INTERFACE_BUILDER` (which we can’t use) and `-prepareForInterfaceBuilder`. Here’s more information about designables, in two flavors: [corporate](https://developer.apple.com/library/ios/recipes/xcode_help-IB_objects_media/Chapters/CreatingaLiveViewofaCustomObject.html), [hipster](http://nshipster.com/ibinspectable-ibdesignable/).

In NSProcessInfo+MGLAdditions, I added a category method that cheaply determines whether the `MGLMapView` is being run inside of IBDesignablesAgentCocoaTouch. I’m using it here to prevent the shared `MGLMapboxEvents` instance from ever being created. The assumption is that you’d never release an app named IBDesignablesAgentCocoaTouch. :smile: